packages/flutter_tools/test/general.shard/compile_test.dart

```diff
@@ -12,14 +14,32 @@ import '../src/common.dart';
 
 void main() {
   testWithoutContext('StdoutHandler can produce output message', () async {
-    final StdoutHandler stdoutHandler = StdoutHandler(logger: BufferLogger.test());
+    final StdoutHandler stdoutHandler = StdoutHandler(logger: BufferLogger.test(), fileSystem: MemoryFileSystem.test());
     stdoutHandler.handler('result 12345');
     expect(stdoutHandler.boundaryKey, '12345');
     stdoutHandler.handler('12345');
     stdoutHandler.handler('12345 message 0');
     final CompilerOutput output = await stdoutHandler.compilerOutput.future;
     expect(output.errorCount, 0);
     expect(output.outputFilename, 'message');
+    expect(output.data, null);
+  });
+
+  testWithoutContext('StdoutHandler can read output bytes', () async {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    final StdoutHandler stdoutHandler = StdoutHandler(logger: BufferLogger.test(), fileSystem: fileSystem);
+    fileSystem.file('message').writeAsBytesSync(<int>[1, 2, 3 ,4]);
+
+    stdoutHandler.reset(readFile: true);
+    stdoutHandler.handler('result 12345');
+    expect(stdoutHandler.boundaryKey, '12345');
+    stdoutHandler.handler('12345');
+    stdoutHandler.handler('12345 message 0');
+    final CompilerOutput output = await stdoutHandler.compilerOutput.future;
+
+    expect(output.errorCount, 0);
```

Comment:
Should there be a test for error count != 0?

Reply:
actually - do we generate an intentionally invalid dill for expression evaluation if there is a compile error? @aam

Reply:
There will be a dill file even if there were compilation errors yes

Reply:
OK - then we should definitely still read it so that the IDE can get the appropriate compilation errors from the attached device